src/cpp/src/llm_pipeline_stateful.cpp

```diff
@@ -161,6 +163,15 @@ DecodedResults StatefulLLMPipeline::generate(
         }
     }
 
+    if (m_is_npu) {
+        // Prefill model in NPU is reshaped to NPUW_LLM_MAX_PROMPT_LEN x NPUW_LLM_MAX_PROMPT_LEN
+        if (encoded_input.input_ids.get_size() > m_max_prompt_len) {
+            OPENVINO_THROW("Stateful LLM pipeline on NPU may only process prompts or hold "
```

Comment:
These two `if`s can be merged into a single one.
There's `OPENVINO_ASSERT` which combines condition check and `OPENVINO_THROW`. Use it instead.

Reply:
Left one `if` and used `OPENVINO_ASSERT`